{"CVE Code": "CVE-2024-23752", "Description": "GenerateSDFPipeline in synthetic_dataframe in PandasAI (aka pandas-ai) through 1.5.17 allows attackers to trigger the generation of arbitrary Python code that is executed by SDFCodeExecutor. An attacker can create a dataframe that provides an English language specification of this Python code. NOTE: the vendor previously attempted to restrict code execution in response to a separate issue, CVE-2023-39660.", "NVD Link": "https://nvd.nist.gov/vuln/detail/CVE-2024-23752", "Vulnerability Categories": "N/A", "Affected Products": "\nGabrieleventuri\u00bbPandasai\u00bbForPythonVersions  up to, including, (<=) 1.5.17cpe:2.3:a:gabrieleventuri:pandasai:*:*:*:*:*:python:*:*Matching versions", "EPSS Score": "0.14% Probability of exploitation activity in the next 30 days\n~ 50 % Percentile, the proportion of vulnerabilities that are scored at or less", "EPSS History": "\n #  Date  Old EPSS Score  New EPSS Score  Delta (New - Old) \n 1  2024-01-30  0.04%  0.14%  +0.10 ", "CVSS Scores": "9.8", "CWE": [{"CWE_ID": "CWE-862 Missing Authorization", "CWE_link": "/cwe-details/862/Missing-Authorization.html"}], "Reference": [{"ref_link": "https://github.com/gventuri/pandas-ai/issues/868", "ref_desc": "[Vulnerability] synthetic_dataframe has serious prompt injection issue which could lead to arbitary command execution \u00b7 Issue #868 \u00b7 gventuri/pandas-ai \u00b7 GitHubExploit;Mailing List;Vendor Advisory", "ref_cve_link": "N/A"}]}